airbyte-integrations/bases/base-normalization/normalization/transform_catalog/transform.py

```diff
@@ -222,275 +452,149 @@ def quote(input_name: str, integration_type: str, in_jinja=False) -> str:
     return result
 
 
-def json_extract_base_property(path: List[str], json_col: str, name: str, definition: dict, integration_type: str) -> Optional[str]:
-    current = path + [name]
+def find_properties_object(path: List[str], field: str, properties, integration_type: str) -> dict:
+    result = {}
+    current_path = path + [field]
+    current = "_".join(current_path)
+    if isinstance(properties, str) or isinstance(properties, int):
+        return {}
+    else:
+        if "items" in properties:
+            return find_properties_object(path, field, properties["items"], integration_type=integration_type)
+        elif "properties" in properties:
+            # we found a properties object
+            return {current: properties["properties"]}
+        elif "type" in properties and is_simple_property(properties["type"]):
+            # we found a basic type
+            return {current: None}
+        elif isinstance(properties, dict):
+            for key in properties.keys():
+                child = find_properties_object(path=current_path, field=key, properties=properties[key], integration_type=integration_type)
+                if child:
+                    result.update(child)
+        elif isinstance(properties, list):
+            for item in properties:
+                child = find_properties_object(path=current_path, field=field, properties=item, integration_type=integration_type)
+                if child:
+                    result.update(child)
+    return result
+
+
+def json_extract_property(json_col: str, name: str, definition: dict, integration_type: str) -> Optional[str]:
+    current = [name]
     if "type" not in definition:
-        return None
-    elif is_string(definition["type"]):
+        return "{} as {}".format(jinja_call(f"json_extract({json_col}, {current})"), quote(name, integration_type))
+    elif is_array(definition["type"]):
+        return "{} as {}".format(jinja_call(f"json_extract_array({json_col}, {current})"), quote(name, integration_type))
+    elif is_object(definition["type"]):
+        return "{} as {}".format(jinja_call(f"json_extract({json_col}, {current})"), quote(name, integration_type))
+    elif is_simple_property(definition["type"]):
```

Comment:
why is string handled separately from the rest of the simple properties? it looks like they are treated the same?

Reply:
> why is string handled separately from the rest of the simple properties? it looks like they are treated the same?

Why do you say string is handled separately? it's part of simple properties and treated the same as int, bool, float. All of these are extracted as "scalar"

The only difference is between those "basic" types and complex ones, it calls different API from warehouses:
- json_extract_scalar: for "simple" types
- json_extract: for non-simple types
- json_extract_array: for arrays

Reply:
huh. i must have misread something here. i think i didn't parse the `elif is_string` as being removed. agreed. you're right.